http/src/main/scala/filodb/http/PrometheusApiRoute.scala

```diff
@@ -157,13 +164,21 @@ class PrometheusApiRoute(nodeCoord: ActorRef, settings: HttpSettings)(implicit a
   }
 
   private def askQueryAndRespond(dataset: String, logicalPlan: LogicalPlan, explainOnly: Boolean, verbose: Boolean,
-                                 spread: Option[Int], tsdbQueryParams: TsdbQueryParams, histMap: Boolean) = {
+                                 spread: Option[Int], tsdbQueryParams: TsdbQueryParams, histMap: Boolean,
+                                 partialResults: Boolean) = {
+    val allowPartialResults = if (partialResults) {
+      if (logicalPlan.isInstanceOf[MetadataQueryPlan])
+         queryConfig.allowPartialResultsMetadataQuery
+      else queryConfig.allowPartialResultsRangeQuery
+    } else false
     val spreadProvider: Option[SpreadProvider] = spread.map(s => StaticSpreadProvider(SpreadChange(0, s)))
     val command = if (explainOnly) {
-      ExplainPlan2Query(DatasetRef.fromDotString(dataset), logicalPlan, QueryContext(tsdbQueryParams, spreadProvider))
+      ExplainPlan2Query(DatasetRef.fromDotString(dataset), logicalPlan, QueryContext(tsdbQueryParams, spreadProvider,
+        allowPartialResults))
     }
     else {
-      LogicalPlan2Query(DatasetRef.fromDotString(dataset), logicalPlan, QueryContext(tsdbQueryParams, spreadProvider))
+      LogicalPlan2Query(DatasetRef.fromDotString(dataset), logicalPlan, QueryContext(tsdbQueryParams, spreadProvider,
+        allowPartialResults))
```

Comment:
Can we simply pass the `partialResults` flag as a planner param, and let the planner deal with the logic ? Looks like this logic belongs in the query planner.

Reply:
PartialResults is part of PlannerParams.  Here it is been passed to QueryContext constructor.

Reply:
`case class PlannerParams(applicationId: String = "filodb",
                         spread: Option[Int] = None,
                         spreadOverride: Option[SpreadProvider] = None,
                         shardOverrides: Option[Seq[Int]] = None,
                         targetSchemaProviderOverride: Option[TargetSchemaProvider] = None,
                         queryTimeoutMillis: Int = 30000,
                         sampleLimit: Int = 1000000,
                         groupByCardLimit: Int = 100000,
                         joinQueryCardLimit: Int = 100000,
                         timeSplitEnabled: Boolean = false,
                         minTimeRangeForSplitMs: Long = 1.day.toMillis,
                         splitSizeMs: Long = 1.day.toMillis,
                         skipAggregatePresent: Boolean = false,
                         processFailure: Boolean = true,
                         processMultiPartition: Boolean = false,
                         allowPartialResults: Boolean = false)`